dataactvalidator/interfaces/interfaceHolder.py

```diff
@@ -8,10 +9,17 @@ class InterfaceHolder:
 
     def __init__(self):
         """ Create the interfaces """
-        self.jobDb = ValidatorJobTrackerInterface()
-        self.errorDb = ValidatorErrorInterface()
-        self.stagingDb = ValidatorStagingInterface()
-        self.validationDb = ValidatorValidationInterface()
+        if BaseInterface.interfaces is None:
+            self.jobDb = ValidatorJobTrackerInterface()
+            self.errorDb = ValidatorErrorInterface()
+            self.stagingDb = ValidatorStagingInterface()
+            self.validationDb = ValidatorValidationInterface()
+            BaseInterface.interfaces = self
```

Comment:
I'm not quite clear on how this approach would work, where does the InterfaceHolder get instantiated?  To provide some background, we have found in the past that any time a new thread starts it needs to instantiate a separate InterfaceHolder to avoid conflicts in the session and connection objects.  So InterfaceHolder has needed to be instantiated in other places, and we'd need a way of dealing with that.  We could manually set BaseInterface.interface each time we instantiate a new one, but that seems to have the same problem as doing it in the constructor.

To expand a little on the current architecture, dataactcore is the common package used by dataactbroker and dataactvalidator, which should never reference each other or be referenced from the core.  The broker app uses only the core and broker, and the validator app uses only the core and validator, so neither should be accessing both InterfaceHolders.  These originally lived in different repos, these days the separation is only enforced by usage, so I agree that finding a way to better integrate the two is a good idea, I'm just not sure how to apply this approach. We could go the other direction and completely separate them, with BaseInterface having both a brokerInterface and a validatorInterface property, with each app only referring to the corresponding property.  That would resolve the immediate concern of a conflict between the two being possible.